src/object/obj_tx.c

```diff
@@ -2072,6 +2073,7 @@ dc_tx_add_update(struct dc_tx *tx, daos_handle_t oh, uint64_t flags,
 				daos_iov_free(&iod_array->oia_iods[i].iod_name);
```

Comment:
Sorry, it is not your patch issue, but here, we need to check "if (iod_array->oia_iods != NULL)" instead of "if (iod_array != NULL)". Similarly for above fixes, also needs to check "if (iod_array->oia_iods != NULL)" during cleanup.

Reply:
Done.  I'm slightly surprised that coverity didn't pick this up but I've made the change.

There is one coverity issue in this file about access of dtrgs when NULL on L1635 but I'll leave that for another PR.